docs/graph/graph-matching/general-match.md

```diff
@@ -216,201 +218,474 @@ vector<int> find_max_unweighted_matching(const undirectedgraph<T> &g) {
-          if (match[i] == -1) {
-            for (auto id : g.g[i]) {
-              auto &e = g.edges[id];
-              int to = e.from ^ e.to ^ i;
-              if (match[to] == -1) {
-                match[i] = to;
-                match[to] = i;
-                break;
-              }
-            }
+      }
+    }
+    return false;
+  };  // bfs
+
+  auto greedy = [&]() {
+    vector<int> order(g.n);
+    // 随机打乱 order
+    iota(order.begin(), order.end(), 0);
+    shuffle(order.begin(), order.end(), rng);
+
+    // 将可以匹配的点匹配
+    for (int i : order) {
+      if (match[i] == -1) {
+        for (auto id : g.g[i]) {
+          auto &e = g.edges[id];
+          int to = e.from ^ e.to ^ i;
+          if (match[to] == -1) {
+            match[i] = to;
+            match[to] = i;
+            break;
           }
         }
-      };  // greedy
-    
-      // 一开始先随机匹配
-      greedy();
-      // 对未匹配点找增广路
-      for (int i = 0; i < g.n; i++) {
-        if (match[i] == -1) {
-          bfs(i);
-        }
       }
-      return match;
     }
-    
-    int main() {
-      ios::sync_with_stdio(0), cin.tie(0);
-      int n, m;
-      cin >> n >> m;
-      undirectedgraph<int> g(n);
-      int u, v;
-      for (int i = 0; i < m; i++) {
-        cin >> u >> v;
-        u--;
-        v--;
-        g.add(u, v, 1);
-      }
-      auto blossom_match = find_max_unweighted_matching(g);
-      vector<int> ans;
-      int tot = 0;
-      for (int i = 0; i < blossom_match.size(); i++) {
-        ans.push_back(blossom_match[i]);
-        if (blossom_match[i] != -1) {
-          tot++;
+  };  // greedy
+
+  // 一开始先随机匹配
+  greedy();
+  // 对未匹配点找增广路
+  for (int i = 0; i < g.n; i++) {
+    if (match[i] == -1) {
+      bfs(i);
+    }
+  }
+  return match;
+}
+
+int main() {
+  ios::sync_with_stdio(0), cin.tie(0);
+  int n, m;
+  cin >> n >> m;
+  undirectedgraph<int> g(n);
+  int u, v;
+  for (int i = 0; i < m; i++) {
+    cin >> u >> v;
+    u--;
+    v--;
+    g.add(u, v, 1);
+  }
+  auto blossom_match = find_max_unweighted_matching(g);
+  vector<int> ans;
+  int tot = 0;
+  for (int i = 0; i < blossom_match.size(); i++) {
+    ans.push_back(blossom_match[i]);
+    if (blossom_match[i] != -1) {
+      tot++;
+    }
+  }
+  cout << (tot >> 1) << "\n";
+  for (auto x : ans) {
+    cout << x + 1 << " ";
+  }
+}
+```
+
+## 基于高斯消元的一般图匹配算法
+
+??? tip "提示"
+
+在阅读以下内容前，你可能需要先阅读“线性代数”部分中关于矩阵的内容：
+
+- [矩阵](/docs/math/linear-algebra/matrix.md)
+- [行列式](/docs/math/linear-algebra/determinant.md)
+- [高斯消元](/docs/math/linear-algebra/gauss.md)
+
+这一部分将介绍一种基于高斯消元的一般图匹配算法。与传统的带花树算法相比，它的优势在于更易于理解与编写，同时便于解决“最大匹配中的必须点”等问题；缺点在于常数比较大，因为高斯消元的 $O(n^3)$ 基本是跑满的，而带花树一般跑不满。
+
+### 前置知识：Tutte 矩阵
+
+**定义**：对于一张 $n$ 个点的无向图 $G = (V, E)$，其 Tutte 矩阵 $\tilde{A}(G)$ 为一个 $n \times n$ 的矩阵，其中：
+
+$\tilde{A}(G)_{i,j} = \left\{ \begin{aligned} x_{i,j} & \quad & i<j,\; (v_i, v_j)\in E \\ -x_{i,j} & \quad & i > j,\; (v_i, v_j) \in E \\ 0 & \quad & \text{otherwise}\end{aligned} \right.$
```

Comment:
```suggestion
$$
\tilde{A}(G)_{i,j} = \left\{ \begin{aligned} x_{i,j} & \quad & i<j,\; (v_i, v_j)\in E \\ -x_{i,j} & \quad & i > j,\; (v_i, v_j) \in E \\ 0 & \quad & \text{otherwise}\end{aligned} \right.
$$
```

Reply:
Applied, many thanks.